plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -741,7 +742,7 @@ private List<Partition> getExistingPartitionsByNames(Table table, List<String> p
     @Override
     public Optional<Partition> getPartition(Table table, List<String> partitionValues)
     {
-        return get(partitionCache, hivePartitionName(hiveTableName(table.getDatabaseName(), table.getTableName()), partitionValues));
+        return get(partitionCache, hivePartitionName(hiveTableName(table.getDatabaseName(), table.getTableName()), partitionValues), () -> delegate.getPartition(table, partitionValues));
```

Comment:
Is this related to "Disallow using of single-entry loader in bulk cache"?

Reply:
> Is this related to "Disallow using of single-entry loader in bulk cache"?

It is, since single-entry loader (in `LoadincCache`) is not allowed, we need to provide explicit loader here.

Reply:
We have two bulk caches: partitionCache and partitionStatisticsCache.

I understand that partitionStatisticsCache is used in bulk mode only, so we may want to disable single-value loader for it, to reduce amount of code.

partitionCache is accessed in bulk and not bulk way, so forcing it to bulk-only and then workaround the just-introduced limitation doesn't seem to be fruitful

What about simplifying this change to just  the following?

```diff
--- plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java
+++ plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java
@@ -271,7 +271,12 @@ public class CachingHiveMetastore
         grantedPrincipalsCache = cacheFactory.buildCache(this::loadPrincipals);
         configValuesCache = cacheFactory.buildCache(this::loadConfigValue);

-        partitionStatisticsCache = partitionCacheFactory.buildCache(this::loadPartitionColumnStatistics, this::loadPartitionsColumnStatistics);
+        partitionStatisticsCache = partitionCacheFactory.buildCache(
+                key -> {
+                    // Single value loader is not used
+                    throw new UnsupportedOperationException();
+                },
+                this::loadPartitionsColumnStatistics);
         partitionFilterCache = partitionCacheFactory.buildCache(this::loadPartitionNamesByFilter);
         partitionCache = partitionCacheFactory.buildCache(this::loadPartitionByName, this::loadPartitionsByNames);
     }
@@ -406,17 +411,6 @@ public class CachingHiveMetastore
                 .collect(toImmutableMap(entry -> entry.getKey().getPartitionName().orElseThrow(), Entry::getValue));
     }

-    private PartitionStatistics loadPartitionColumnStatistics(HivePartitionName partition)
-    {
-        HiveTableName tableName = partition.getHiveTableName();
-        String partitionName = partition.getPartitionName().orElseThrow();
-        Table table = getExistingTable(tableName.getDatabaseName(), tableName.getTableName());
-        Map<String, PartitionStatistics> partitionStatistics = delegate.getPartitionStatistics(
-                table,
-                ImmutableList.of(getExistingPartition(table, partition.getPartitionValues())));
-        return partitionStatistics.get(partitionName);
-    }
```

Reply:
```
+        partitionStatisticsCache = partitionCacheFactory.buildCache(
+                key -> {
+                    // Single value loader is not used
+                    throw new UnsupportedOperationException();
+                },
+                this::loadPartitionsColumnStatistics);
```
is not needed as `buildBulkCache` will throw exception when single entry loader is used

```
-    private PartitionStatistics loadPartitionColumnStatistics(HivePartitionName partition)
-    {
-        HiveTableName tableName = partition.getHiveTableName();
-        String partitionName = partition.getPartitionName().orElseThrow();
-        Table table = getExistingTable(tableName.getDatabaseName(), tableName.getTableName());
-        Map<String, PartitionStatistics> partitionStatistics = delegate.getPartitionStatistics(
-                table,
-                ImmutableList.of(getExistingPartition(table, partition.getPartitionValues())));
-        return partitionStatistics.get(partitionName);
-    }
```
it's already removed

Reply:
> partitionCache is accessed in bulk and not bulk way, so forcing it to bulk-only and then workaround the just-introduced limitation doesn't seem to be fruitful

It's not a workaround. It reduces number of metastore calls when metadata cache is disabled after https://github.com/trinodb/trino/pull/15811. It also makes processing streamlined.